src/poll.rs

```diff
@@ -468,13 +470,25 @@ mod notify {
     impl Notify {
         /// Creates a new notification pipe.
         pub(super) fn new() -> io::Result<Self> {
+            #[cfg(not(target_os = "haiku"))]
             let (read_pipe, write_pipe) = pipe_with(PipeFlags::CLOEXEC).or_else(|_| {
                 let (read_pipe, write_pipe) = pipe()?;
                 fcntl_setfd(&read_pipe, fcntl_getfd(&read_pipe)? | FdFlags::CLOEXEC)?;
                 fcntl_setfd(&write_pipe, fcntl_getfd(&write_pipe)? | FdFlags::CLOEXEC)?;
                 io::Result::Ok((read_pipe, write_pipe))
             })?;
 
+            #[cfg(target_os = "haiku")]
+            let pipe_with = None;
+
+            #[cfg(target_os = "haiku")]
+            let (read_pipe, write_pipe) = pipe_with.unwrap_or_else(|| {
+                let (read_pipe, write_pipe) = pipe()?;
+                fcntl_setfd(&read_pipe, fcntl_getfd(&read_pipe)? | FdFlags::CLOEXEC)?;
+                fcntl_setfd(&write_pipe, fcntl_getfd(&write_pipe)? | FdFlags::CLOEXEC)?;
+                io::Result::Ok((read_pipe, write_pipe))
+            })?;
+
```

Comment:
I'd rather not duplicate the pipe creation code twice. Instead, can you make a closure that creates the pipe, then call it in the Haiku case and use it in `or_else` in the other case?

Reply:
updated